pkg/query-service/app/http_handler.go

```diff
@@ -5650,3 +5659,535 @@ func (aH *APIHandler) getDomainInfo(w http.ResponseWriter, r *http.Request) {
+		aH.handleSlowTraces(w, r, true)
+	}).Methods("POST")
+
+}
+
+// handleNewFunnel creates a new funnel without steps
+// Steps should be added separately using the update endpoint
+func (aH *APIHandler) handleNewFunnel(w http.ResponseWriter, r *http.Request) {
+	var req traceFunnels.NewFunnelRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	claims, ok := authtypes.ClaimsFromContext(r.Context())
+	if !ok {
+		http.Error(w, "unauthenticated", http.StatusUnauthorized)
+		return
+	}
+	userID := claims.UserID
+	orgID := claims.OrgID
+
+	// Validate timestamp is provided and in milliseconds format
+	if err := traceFunnels.ValidateTimestamp(req.Timestamp, "creation_timestamp"); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Check for name collision in the SQLite database
+	exists, err := aH.TraceFunnels.CheckFunnelNameCollision(req.Name, userID)
+	if err != nil {
+		zap.L().Error("Error checking for funnel name collision: %v", zap.Error(err))
+		http.Error(w, fmt.Sprintf("failed to check funnel name collision: %v", err), http.StatusInternalServerError)
+		return
+	} else if exists {
+		http.Error(w, fmt.Sprintf("funnel with name '%s' already exists for user '%s' in the database", req.Name, userID), http.StatusBadRequest)
+		return
+	}
+
+	funnel := &traceFunnels.Funnel{
+		ID:        uuid.New().String(),
+		Name:      req.Name,
+		CreatedAt: req.Timestamp * 1000000, // Convert milliseconds to nanoseconds for internal storage
+		CreatedBy: userID,
+		OrgID:     orgID,
+		Steps:     make([]traceFunnels.FunnelStep, 0),
+	}
+
+	// Create new funnel in database
+	err = aH.TraceFunnels.CreateFunnel(funnel, userID, orgID)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to save funnel to database: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	response := traceFunnels.NewFunnelResponse{
+		ID:        funnel.ID,
+		Name:      funnel.Name,
+		CreatedAt: funnel.CreatedAt / 1000000,
+		CreatedBy: funnel.CreatedBy,
+		OrgID:     orgID,
+	}
+	json.NewEncoder(w).Encode(response)
+}
+
+// handleUpdateFunnelStep adds or updates steps for an existing funnel
+// Steps are identified by their step_order, which must be unique within a funnel
+func (aH *APIHandler) handleUpdateFunnelStep(w http.ResponseWriter, r *http.Request) {
+	var req traceFunnels.FunnelStepRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	claims, ok := authtypes.ClaimsFromContext(r.Context())
+	if !ok {
+		http.Error(w, "unauthenticated", http.StatusUnauthorized)
+		return
+	}
+	userID := claims.UserID
+
+	if err := traceFunnels.ValidateTimestamp(req.Timestamp, "updated_timestamp"); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	//aH.Signoz.SQLStore.SQLxDB()
+	dbClient, _ := traceFunnels.NewSQLClient(aH.Signoz.SQLStore)
```

Comment:
why is db client present here ?

Reply:
using the client from aH.TraceFunnels throughout, missed this during the refactor, thanks for pointing out